test/functional/test_framework/test_node.py

```diff
@@ -97,14 +97,16 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
         ]
-        if use_valgrind:
+        if self.is_valgrind_present() and use_valgrind:
```

Comment:
This will create an unnecessary `valgrind` execution in the normal case when Valgrind is not used (no `--valgrind`).

Better to do it as `if use_valgrind and …` instead :)

Reply:
Yes, indeed. Thanks!